### PR TITLE
Fixing substore notation in module LIPs

### PR DESCRIPTION
**Pull request by AnanyaSshri**
This PR fixes the formulation of substore specifications in the "Notation" bullet point. 

In particular, <substore name> entry is treated like an object although it of type bytes since it is the serialization of an object. So we are reformulating it by adding that it will be deserialized using that schema.

### CODE REVIEW

**Comment by mjerkov** [comment]
Looks good to me, waiting for @sergeyshemyakov's comment...

**Review by mjerkov — APPROVED** [review]

**Comment by mjerkov** [comment]
As I see, there some other substores, in LIP44, LIP47 etc., so I suppose you're still working on it (that's I first approved, and then self re-requested the review)...

**Comment by AnanyaSshri** [comment]
> As I see, there are some other substores, in LIP44, LIP47 etc., so I suppose you're still working on it (that's I first approved, and then self re-requested the review)...

I have done the changes pointed in LIP44 and LIP47 specifically. You can re-review now.

File: proposals/lip-0044.md [hunk 1]
```diff
@@ -208,7 +208,7 @@ This substore stores the current validator parameters as set using the [setValid
 * The substore prefix is set to `SUBSTORE_PREFIX_BLS_KEYS`.
 * Store keys are of type `PublicKeyBLS`.
 * Store values are set to the addresses of the validators corresponding to the store keys, serialized using the `validatorAddressSchema` schema presented [below](#json-schema-2).
-* Notation: For the rest of this proposal let `registeredBLSKeys(blsKey)` be the entry in the registered BLS keys substore identified by the store key `blsKey`.
+* Notation: For the rest of this proposal let `registeredBLSKeys(blsKey)` be the entry in the registered BLS keys substore identified by the store key `blsKey`, `validatorAddressSchema` schema.
```
Comment (mjerkov) [inline proposals/lip-0044.md:211]:
```suggestion
* Notation: For the rest of this proposal let `registeredBLSKeys(blsKey)` be the entry in the registered BLS keys substore identified by the store key `blsKey`, deserialized using `validatorAddressSchema` schema.
```

File: proposals/lip-0047.md [hunk 2]
```diff
@@ -141,7 +141,7 @@ The name substore is an auxiliary store used to validate the [register authority
 * The substore prefix is set to `SUBSTORE_PREFIX_NAME`.
 * Each substore key is a name of a validator as given in the genesis block or with an [register authority command](#register-authority-command), serialized as a utf-8 encoded string.
 * Each substore value is set to the address of the corresponding validator, serialized according to the JSON schema `validatorAddressSchema` below.
-* Notation: let `name(nameBytes)` denote the entry in the name substore with the key `nameBytes`.
+* Notation: let `name(nameBytes)` denote the entry in the name substore with the key `nameBytes`, deserialized using the `validatorAddressSchema` schema..
```
Comment (mjerkov) [inline proposals/lip-0047.md:144]:
```suggestion
* Notation: let `name(nameBytes)` denote the entry in the name substore with the key `nameBytes`, deserialized using the `validatorAddressSchema` schema.
```

File: proposals/lip-0047.md [hunk 3]
```diff
@@ -172,7 +172,7 @@ This substore contains the snapshot of the active authorities for the current ro
 * The substore prefix is set to `SUBSTORE_PREFIX_SNAPSHOT`.
 * Each substore key is `uint32be(roundNumber)`, where `roundNumber` can be 0, 1 or 2 corresponding to the current round, the next round and in two rounds respectively.
 * Each substore value is the serialization of an object following `snapshotStoreSchema`.
-* Notation: Let `snapshotStore(roundNumber)` be the substore entry with prefix `SUBSTORE_PREFIX_SNAPSHOT` and key `uint32be(roundNumber)`.
+* Notation: Let `snapshotStore(roundNumber)` be the substore entry with prefix `SUBSTORE_PREFIX_SNAPSHOT`, deserialized using the `snapshotStoreSchema` schema,  and key `uint32be(roundNumber)`.
```
Comment (mjerkov) [inline proposals/lip-0047.md:175]: To be consistent with `Notation` part for other substores:
```suggestion
* Notation: Let `snapshotStore(roundNumber)` be the substore entry in the snapshot substore with the key `uint32be(roundNumber)`, deserialized using the `snapshotStoreSchema` schema.
```

**Review by mjerkov — COMMENTED** [review]
I added a few small comments, i.e., suggested changes.

File: proposals/lip-0044.md (hunk 1 above)
Comment (AnanyaSshri) [reply proposals/lip-0044.md]: Done

File: proposals/lip-0047.md (hunk 2 above)
Comment (AnanyaSshri) [reply proposals/lip-0047.md]: Done

File: proposals/lip-0047.md (hunk 3 above)
Comment (AnanyaSshri) [reply proposals/lip-0047.md]: Done

**Review by sergeyshemyakov — CHANGES_REQUESTED** [review]
Some more things to address:

1. In LIP 0046 in [this section](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0046.md#substore-prefix-store-key-and-store-value) I would add:
> Notation: For the rest of this proposal let `validatorReveals` be the entry in the random substore, deserialized using `seedRevealSchema` schema.

Note that the notation `validatorReveals` is already used in the LIP (e.g. in [this function](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0046.md#execution-1))
2. In LIP 0057 in [the previous timestamp store](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0057.md#store-prefix-store-key-and-store-value-6) we don't say "deserialized using...".
3. In LIP 0058 we don't say "deserialized using...".
4. In [LIP 0071](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0071.md#substore-prefix-store-key-and-store-value) we don't say "deserialized using...".

**Review by sergeyshemyakov — APPROVED** [review]
Congrats on your first PR!

**Review by janhack — APPROVED** [review]
Editorial approval

**Pull request merged**